pkg/execution/state/redis_state/queue.go

```diff
@@ -377,6 +381,36 @@ type QueueItem struct {
 	QueueName *string `json:"queueID,omitempty"`
 }
 
+// Score returns the score (time that the item should run) for the queue item.
+//
+// NOTE: In order to prioritize finishing older function runs with a busy function
+// queue, we sometimes use the function run's "started at" time to enqueue edges which
+// run steps.  This lets us push older function steps to the beginning of the queue,
+// ensuring they run before other newer function runs.
+//
+// We can ONLY do this for the first attempt, and we can ONLY do this for edges that
+// are not sleeps (eg. immediate runs)
```

Comment:
why only the first attempt?

Reply:
Because attempts >= 1 are scheduled in the future.